src/BuildScriptGeneratorCli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,109 @@
+        "with the best-matching platform tool versions.")]
+    internal class ExecCommand : CommandBase
+    {
+        public const string Name = "exec";
+
+        [Option("-s|--src <dir>", CommandOptionType.SingleValue, Description = "Source directory.")]
+        [DirectoryExists]
+        public string SourceDir { get; set; }
+
+        [Argument(1, Description = "The command to execute in an app-specific environment.")]
+        public string Command { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<ExecCommand>>();
+            var env = serviceProvider.GetRequiredService<IEnvironment>();
+            var generator = serviceProvider.GetRequiredService<IBuildScriptGenerator>();
+
+            if (string.IsNullOrWhiteSpace(Command))
+            {
+                logger.LogDebug("Command is empty; exiting");
+                return ProcessConstants.ExitSuccess;
+            }
+
+            var shellPath = env.GetEnvironmentVariable("BASH") ?? FilePaths.Bash;
+            logger.LogInformation("Using shell {shell}", shellPath);
+
+            var ctx = BuildScriptGenerator.CreateContext(serviceProvider, operationId: null);
+            ctx.DisableMultiPlatformBuild = false;
+            var tools = generator.GetRequiredToolVersions(ctx);
+
+            if (tools.Count == 0)
+            {
+                console.WriteErrorLine(Labels.ExecCommandNoToolsDetectedErrorMessage);
+                return ProcessConstants.ExitFailure;
+            }
+
+            int exitCode;
+            using (var timedEvent = logger.LogTimedEvent("ExecCommand"))
+            {
+                var benvCmd = $"{FilePaths.Benv} {StringExtensions.JoinKeyValuePairs(tools)}";
+
+                // Build envelope script
+                var script = new ShellScriptBuilder("\n")
+                    .AddShebang(shellPath)
+                    .Source(benvCmd)
+                    .AddCommand(Command)
+                    .ToString();
+                logger.LogDebug("Script content:\n{script}", script);
+
+                // Create temporary file to store script
+                var tempScriptPath = Path.GetTempFileName();
+                File.WriteAllText(tempScriptPath, script);
+                timedEvent.AddProperty(nameof(tempScriptPath), tempScriptPath);
+
+                if (DebugMode)
+                {
+                    console.WriteLine($"Temporary script @ {tempScriptPath}:");
+                    console.WriteLine("---");
+                    console.WriteLine(script);
+                    console.WriteLine("---");
+                }
+
+                exitCode = ProcessHelper.RunProcess(
+                    shellPath,
+                    new[] { tempScriptPath },
+                    serviceProvider.GetRequiredService<IOptions<BuildScriptGeneratorOptions>>().Value.SourceDir,
```

Comment:
nit: get the options into a variable (after line 36) and use it here instead.

Reply:
that's the purpose of this? I preferred to have one less variable because this isn't used anywhere else

Reply:
It just looks ugly IMO. You could argue the same thing for a `using` statement. We do not put the whole namespace at the point we use because we use it in one place.